### ISSUE TITLE
calling `EguiContext::ctx` from a startup system

### ISSUE THREAD

**Issue by RichMQ** [mvlabat/bevy_egui#41]
I would like to globally modify some egui visuals, I tried using this
```
fn ui_setup(egui_context: ResMut<EguiContext>) {
    egui_context.ctx().set_visuals(egui::Visuals {
        window_shadow: Default::default(),
        ..Default::default()
    });
}
```
as a startup system. (Just added this to the `simple.rs` example.) Unfortunately, this results in a panic:
```
thread 'main' panicked at '`EguiContext::ctx()` called before the ctx has been initialized. Consider moving your UI system to `CoreStage::Update` or run you system after `EguiSystem::BeginFrame`.'
```
The message here seems to be relevant to regular systems, but not to startup systems. How do I ensure that my startup system runs after the context has been initialized?

**Comment by mvlabat** [comment]
Hi! Thanks for creating the issue. Tbh, I'm not sure if it's possible to do in a startup system... If it's not, it is definitely an oversight.

Currently, context gets initialised in the `process_input` system, which runs during `CoreStage::PreUpdate`, so the workaround would be to declare a regular system that would run only once (after `CoreStage::PreUpdate` stage) like this:
```
fn ui_setup(mut has_been_run: Local<bool>, egui_context: ResMut<EguiContext>) {
    if *has_been_run {
      return;
    }
    *has_been_run = true;
    egui_context.ctx().set_visuals(egui::Visuals {
        window_shadow: Default::default(),
        ..Default::default()
    });
}
```

Probably, another alternative, that I haven't explored yet, would be just doing `add_startup_system_to_stage(CoreStage::Update , ui_setup.system())`. If it does work, I guess we can mark it as a solution and and document it and add to the examples.

Please let me know if any of the solutions works (I guess the latter is a more preferable one). And thanks for bringing this up!

**Comment by RichMQ** [comment]
Thanks for a reply. Your first suggestion occurred to me, but it is less than ideal to use a normal system as a startup system. Your second suggestion, unfortunately, panics:
```
thread 'main' panicked at 'Stage 'Update' does not exist or is not a SystemStage', ~\.cargo\registry\src\github.com-1ecc6299db9ec823\bevy_ecs-0.5.0\src\schedule\mod.rs:150:17
```
I guess you just can't mix startup systems with `CoreStage`. I have only just started poking at bevy, so I don't really understand the architecture, but wouldn't the right solution be for `bevy_egui` to do its initialization as a startup system with some explicit stage/label that I could refer to while adding my own startup system?

**Comment by RichMQ** [comment]
My previous comment notwithstanding, here's a different workaround which at least doesn't feel as hacky.
```Rust
use bevy::prelude::*;
use bevy_egui::{egui, EguiContext, EguiPlugin};

fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .add_plugin(EguiPlugin)
        .add_state(AppState::UiSetup)
        .add_system_set(
            SystemSet::on_update(AppState::UiSetup)
                .with_system(ui_setup.system())
        )
        .add_system(ui_example.system())
        .run();
}

#[derive(Clone, Debug, Hash, PartialEq, Eq)]
enum AppState {
    UiSetup,
    Running,
}

fn ui_setup(egui_context: ResMut<EguiContext>, mut app_state: ResMut<State<AppState>>) {
    egui_context.ctx().set_visuals(egui::Visuals {
        window_shadow: Default::default(),
        ..Default::default()
    });
    app_state.set(AppState::Running).unwrap();
}

fn ui_example(egui_context: Res<EguiContext>) {
    egui::Window::new("Hello").show(egui_context.ctx(), |ui| {
        ui.label("world");
    });
}
```

**Comment by mvlabat** [comment]
Thank you for taking time to try all the solutions!

> but wouldn't the right solution be for `bevy_egui` to do its initialization as a startup system with some explicit stage/label that I could refer to while adding my own startup system?

Yeah, it totally makes sense, that indeed would be the right solution. While it's possible to workaround the issue, configuring UI during the startup is the most intuitive way, and I'd also like it to work like that.
I'll see if I can fix it. I hope that the context doesn't depend on anything that isn't available during the startup stage.

**Comment by mvlabat** [comment]
Hey @RichMQ! I've just published a PR that should fix the problem: #43. Could you check if it works for you?

You can add the following to your `Cargo.toml` to easily do this:
```
[patch.crates-io]
bevy_egui = { git = "https://github.com/mvlabat/bevy_egui.git", branch = "startup-init" }
```

I'll publish a new release today or tomorrow.

**Comment by mvlabat** [comment]
I've just published a new release, the problem should be fixed now. If you stumble upon any other one, feel free to open a new issue.

**Comment by RichMQ** [comment]
I only just got a chance to check this out. Works fine, thanks!
